### ISSUE TITLE
Device selection (multi GPU)

### ISSUE THREAD

**Issue by tiko13** [city96/ComfyUI-GGUF#96]
Would it be possible to add a device selection into loaders, similar to to https://github.com/neuratech-ai/ComfyUI-MultiGPU

I have tried to 'port' that feature using similar code, but I have not been successful:
even when device is correctly selected as cuda:0, for some reason it is attempting to load gguf models onto cuda:1

torch.cuda.set_device(device) also doesn't work, even when hardcoded, 

There is probably something downstream overriding this selection, but I cannot figure out what it is

**Comment by city96** [comment]
>There is probably something downstream overriding this selection, but I cannot figure out what it is

That makes two of us. Doing `CUDA_VISIBLE_DEVICES=1` just dumps the T5 model onto the CPU for some reason on my P40 though I haven't had time to investigate why this happens (This only happens with the gguf dual clip loader and not the normal one.)

Weirder still, my [crappy bootleg offload script](https://gist.github.com/city96/30743dfdfe129b331b5676a79c3a8a39) seems to work fine lol.

**Comment by tiko13** [comment]
I will try to look at it more when I get time, yet another 3090 is headed my way and this is quickly becoming bigger of an issue :D

**Comment by city96** [comment]
Question, can you check if https://github.com/city96/ComfyUI-GGUF/commit/d2aaeb0f138320cb2b1481d00c79ee63d7cfe81b does anything to fix this? You'll have to update ComfyUI too but it does fix the CPU issue at least. Might make it work with that node set you linked above.

**Comment by tiko13** [comment]
I have tried it now (no vram flag), for some reason, the UNET is now always loaded into cuda:1 device, even if i explicitly set it as cuda:0

this causes CLIP and VAE to run in cuda:0 but the UNET will always go into cuda:1 no matter what

**Comment by city96** [comment]
Yeah, that was by last bet lol. Actually, have you tried running with `CUDA_DEVICE_ORDER=PCI_BUS_ID` ? I doubt it'd help but who knows, might be something with the device order.

**Comment by tiko13** [comment]
So it's even more complicated than I thought: when freshly starting comfy and setting all nodes to cuda:0, it all goes fine on it. As soon as I load the encoder onto cuda:1, the unet is also forced there. BUT, when I change the clip and all other nodes back to cuda:0, CLIP and Vae load correctly, but the unet will reload onto cuda:1 anyway, and this can be fixed only by restarting comfy

**Comment by amssss0** [comment]
+1
